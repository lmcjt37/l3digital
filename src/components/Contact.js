import React from 'react'
import PropTypes from 'prop-types'
import { StaticQuery, graphql } from 'gatsby'

import Form from 'components/Form'

const Container = ({ data }) => {
    const companyData = data.contentfulCompany
    return (
        <section id="contact">
            <div className="inner">
                <section>
                    <Form />
                </section>
                <section className="split">
                    <section>
                        <div className="contact-method">
                            <span className="icon alt fa-envelope" />
                            <h3>Email</h3>
                            <a href={'mailto:' + companyData.email}>
                                {companyData.email}
                            </a>
                        </div>
                    </section>
                    <section>
                        <div className="contact-method">
                            <span className="icon alt fa-home" />
                            <h3>Address</h3>
                            <span>
                                {companyData.address.line1}
                                <br />
                                {companyData.address.line2}
                                <br />
                                {companyData.address.county}
                                <br />
                                {companyData.address.country}
                                <br />
                                {companyData.address.postcode}
                            </span>
                        </div>
                    </section>
                </section>
            </div>
        </section>
    )
}

const Contact = props => (
    <StaticQuery
        query={graphql`
            {
                contentfulCompany(
                    contentful_id: { eq: "27CTUp0dfpHdukUkA75P4j" }
                ) {
                    email
                    telephone1
                    address {
                        line1
                        line2
                        county
                        country
                        postcode
                    }
                }
            }
        `}
        render={data => <Container data={data} {...props} />}
    />
)

export default Contact

Container.propTypes = {
    data: PropTypes.shape({
        line1: PropTypes.string,
        line2: PropTypes.string,
        county: PropTypes.string,
        country: PropTypes.string,
        postcode: PropTypes.object,
    }),
}
