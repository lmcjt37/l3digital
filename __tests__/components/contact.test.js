import { StaticQuery } from 'gatsby'

import Contact from 'components/Contact'
import contactMock from '__mocks__/contact.mock'

beforeEach(() => {
    StaticQuery.mockImplementationOnce(({ render }) => render(contactMock))
})

describe('Contact tests', () => {
    test('check it renders correctly with snapshot', () => {
        const tree = render(<Contact />)
        expect(tree).toMatchSnapshot()
    })
})
