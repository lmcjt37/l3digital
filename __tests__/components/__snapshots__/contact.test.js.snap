// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact tests check it renders correctly with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        id="contact"
      >
        <div
          class="inner"
        >
          <section>
            <form
              action="#"
              method="post"
            >
              <div
                class="field half first"
              >
                <label
                  for="name"
                >
                  Name
                </label>
                <input
                  id="name"
                  name="name"
                  type="text"
                />
              </div>
              <div
                class="field half"
              >
                <label
                  for="email"
                >
                  Email
                </label>
                <input
                  id="email"
                  name="email"
                  type="text"
                />
              </div>
              <div
                class="field"
              >
                <label
                  for="message"
                >
                  Message
                </label>
                <textarea
                  id="message"
                  name="message"
                  rows="6"
                />
              </div>
              <ul
                class="actions"
              >
                <li>
                  <input
                    class="special"
                    type="submit"
                    value="Send Message"
                  />
                </li>
                <li>
                  <button
                    class="effect"
                    type="reset"
                  >
                    Clear
                  </button>
                </li>
              </ul>
            </form>
          </section>
          <section
            class="split"
          >
            <section>
              <div
                class="contact-method"
              >
                <span
                  class="icon alt fa-envelope"
                />
                <h3>
                  Email
                </h3>
                <a
                  href="mailto:hello@l3digital.co.uk"
                >
                  hello@l3digital.co.uk
                </a>
              </div>
            </section>
            <section>
              <div
                class="contact-method"
              >
                <span
                  class="icon alt fa-phone"
                />
                <h3>
                  Phone
                </h3>
                <a
                  href="tel:0845 111 3333"
                >
                  0845 111 3333
                </a>
              </div>
            </section>
            <section>
              <div
                class="contact-method"
              >
                <span
                  class="icon alt fa-home"
                />
                <h3>
                  Address
                </h3>
                <span>
                  L3 Digital Limited
                  <br />
                  Amazing House
                  <br />
                  Leicestershire
                  <br />
                  United Kingdom
                  <br />
                  LE3 T03
                </span>
              </div>
            </section>
          </section>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      id="contact"
    >
      <div
        class="inner"
      >
        <section>
          <form
            action="#"
            method="post"
          >
            <div
              class="field half first"
            >
              <label
                for="name"
              >
                Name
              </label>
              <input
                id="name"
                name="name"
                type="text"
              />
            </div>
            <div
              class="field half"
            >
              <label
                for="email"
              >
                Email
              </label>
              <input
                id="email"
                name="email"
                type="text"
              />
            </div>
            <div
              class="field"
            >
              <label
                for="message"
              >
                Message
              </label>
              <textarea
                id="message"
                name="message"
                rows="6"
              />
            </div>
            <ul
              class="actions"
            >
              <li>
                <input
                  class="special"
                  type="submit"
                  value="Send Message"
                />
              </li>
              <li>
                <button
                  class="effect"
                  type="reset"
                >
                  Clear
                </button>
              </li>
            </ul>
          </form>
        </section>
        <section
          class="split"
        >
          <section>
            <div
              class="contact-method"
            >
              <span
                class="icon alt fa-envelope"
              />
              <h3>
                Email
              </h3>
              <a
                href="mailto:hello@l3digital.co.uk"
              >
                hello@l3digital.co.uk
              </a>
            </div>
          </section>
          <section>
            <div
              class="contact-method"
            >
              <span
                class="icon alt fa-phone"
              />
              <h3>
                Phone
              </h3>
              <a
                href="tel:0845 111 3333"
              >
                0845 111 3333
              </a>
            </div>
          </section>
          <section>
            <div
              class="contact-method"
            >
              <span
                class="icon alt fa-home"
              />
              <h3>
                Address
              </h3>
              <span>
                L3 Digital Limited
                <br />
                Amazing House
                <br />
                Leicestershire
                <br />
                United Kingdom
                <br />
                LE3 T03
              </span>
            </div>
          </section>
        </section>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
